internal/planner/planner.go

```diff
@@ -1019,15 +1027,17 @@ func (p *Planner) planExprCallValue(value *ast.Term, arity int, operands []*ast.
 				return iter()
 			})
 		})
-	}
 
-	// call: f(x, 1) # caller captures result
-	return p.planCallArgs(operands[:len(operands)-1], 0, nil, func([]ir.Operand) error {
-		p.ltarget = p.newOperand()
-		return p.planTerm(value, func() error {
-			return p.planUnifyLocal(p.ltarget, operands[len(operands)-1], iter)
+	case len(operands) == arity+1: // call: f(x, 1) # caller captures result
+		return p.planCallArgs(operands[:len(operands)-1], 0, nil, func([]ir.Operand) error {
+			p.ltarget = p.newOperand()
+			return p.planTerm(value, func() error {
+				return p.planUnifyLocal(p.ltarget, operands[len(operands)-1], iter)
+			})
 		})
-	})
+	default:
```

Comment:
Nothing of importance going on here -- restructured to switch instead of if; added error for the impossible case.